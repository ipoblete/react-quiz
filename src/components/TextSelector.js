import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({ text, onChange }) {
  return (
    <>
      <label>Text:</label>
      <input type="text" name="text" value={text} onChange={onChange}/>
    </>
  );
}

TextSelector.propTypes = {
  text: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default TextSelector;

