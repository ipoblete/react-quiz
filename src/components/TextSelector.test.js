import React from 'react';
import renderer from 'react-test-renderer';
import TextSelector from './TextSelector';

describe('TextSelector', () => {
  it('matches a snapshot', () => {
    const text = renderer.create(
      <TextSelector />
    ).toJSON();
    expect(text).toMatchSnapshot();
  });
});
