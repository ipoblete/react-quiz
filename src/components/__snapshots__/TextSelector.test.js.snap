// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector matches a snapshot 1`] = `
Array [
  <label>
    Text:
  </label>,
  <input
    name="text"
    type="text"
  />,
]
`;
